tardisbase/testing/regression_comparison/run_tests.py

```diff
@@ -0,0 +1,156 @@
+        elif isinstance(commits_input, int):
+            n = commits_input  
+            commits_input = None
+
+        if commits_input:
+            n = len(commits_input)
+            commits = []
+            for commit_hash in commits_input:
+                try:
+                    commit = tardis_repo.commit(commit_hash)
+                    commits.append(commit)
+                except Exception as e:
+                    print(f"Error finding commit {commit_hash}: {e}")
+                    continue
+        else:
+            commits = list(tardis_repo.iter_commits(branch, max_count=n))
+            commits.reverse()
+    else:
+        commits = list(tardis_repo.iter_commits(branch, max_count=n))
+        commits.reverse()
+
+    processed_commits = []
+    regression_commits = []
+
+    for i, commit in enumerate(commits, 1):
+        print(f"Processing commit {i}/{n}: {commit.hexsha}")
+        tardis_repo.git.checkout(commit.hexsha)
+        tardis_repo.git.reset('--hard')
+        tardis_repo.git.clean('-fd')
+
+        env_name = None
+        if use_conda:
+            # Create unique environment name for this commit
+            env_name = f"tardis-test-{commit.hexsha[:8]}"
+            print(f"Creating conda environment: {env_name}")
+            
+            if not create_conda_env(env_name, lockfile_path, conda_manager):
+                print(f"Failed to create conda environment for commit {commit.hexsha}")
+                continue
+            
+            if not install_tardis_in_env(env_name, tardis_path, conda_manager):
+                print(f"Failed to install TARDIS in environment for commit {commit.hexsha}")
+                continue
+
+        # Prepare pytest command
+        if use_conda and env_name:
+            cmd = [
+                conda_manager, "run", "-n", env_name,
+                "python", "-m", "pytest",
+                test_path,
+                f"--tardis-regression-data={regression_path}",
+                "--generate-reference",
+                "-x",
+                "--disable-warnings"
+            ]
+        else:
+            cmd = [
+                "python", "-m", "pytest",
+                test_path,
+                f"--tardis-regression-data={regression_path}",
+                "--generate-reference",
+                "-x",
+                "--disable-warnings"
+            ]
+        
+        print(f"Running pytest command: {' '.join(cmd)}")
+        try:
+            result = subprocess.run(
+                cmd,
+                check=True,
+                capture_output=True,
+                text=True,
+                cwd=tardis_path 
+            )
+            print("Pytest stdout:")
+            print(result.stdout)
+            print("Pytest stderr:")
+            print(result.stderr)
+
+            if not target_file_path.exists():
+                print(f"Error: HDF5 file {target_file_path} was not generated.")
+                continue
```

Comment:
so I give a target file path and its not generated- then it skips that commit. I dont think we want that right?
Shouldn't target file path be optional?